### PR TITLE
fix(curriculum): too strict tests

### LINKED ISSUE #52892: Tests are to strict for step 103 of platformer game
### Describe the Issue

We had a [camper](https://forum.freecodecamp.org/t/learn-intermediate-oop-by-building-a-platformer-game-step-103/660595/2) unable to pass the test, because the test is expecting parenthesis here

```js
if(isCheckpointCollisionDetectionActive) {
    setTimeout(() => ( checkpointScreen.style.display = "none" ), 2000);
  }
```
But this is valid code and should pass
```js
const showCheckpointScreen = (msg) => {
  checkpointScreen.style.display = "block";
  checkpointMessage.textContent = msg;
  if(isCheckpointCollisionDetectionActive) {
    setTimeout(() => { checkpointScreen.style.display = "none"; }, 2000);
  }
};
```
It looks like the last test is causing issue and needs to be updated.
It should accept any valid answer.

# Note to contributors 

This issue is marked available for contribution. The first comprehensive PR created will be reviewed and merged. We typically do not assign issues to anyone other than long-time contributors.

If you would like to contribute and have not read the contributors docs, please do so here: https://contribute.freecodecamp.org/#/

Here is the file you will need to change
https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb4ebdc75b3a73a43da5ec.md

If you have any issues with contributing, be sure to join us on the [contributors channel](https://discord.com/channels/692816967895220344/715074489422970962), or on the [contributors sub-forum](https://forum.freecodecamp.org/c/contributors/3)

Happy coding! 

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-intermediate-oop-by-building-a-platformer-game/step-103

### Your code

```

const showCheckpointScreen = (msg) => {
  checkpointScreen.style.display = "block";
  checkpointMessage.textContent = msg;
  if(isCheckpointCollisionDetectionActive) {
    setTimeout(() => { checkpointScreen.style.display = "none"; }, 2000);
  }
};

```

### Expected behavior

The code above should pass.
The test is to strict and expects a different result. 

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by usamabasharat**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52892

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb4ebdc75b3a73a43da5ec.md [hunk 1]
```diff
@@ -420,6 +420,9 @@ const startGame = () => {
 const showCheckpointScreen = (msg) => {
   checkpointScreen.style.display = "block";
   checkpointMessage.textContent = msg;
+  if (isCheckpointCollisionDetectionActive) {
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb4ebdc75b3a73a43da5ec.md:423]: It looks like there is a misunderstanding of the issue.
You shouldn't give them the answer.
Also, we don't need to change all of the seed code here.
I would suggest reverting all of your changes to these files.

The issue are the tests for step 103 which can be found in the hints section `# --hints--`

The changes you need to make are to the tests themselves. 

The tests should accept any valid answer but right now they only accept one answer.

Once you have made updates to step 103's tests, make sure to run this locally and test your changes.

Hope that is clearer on what the task is 👍

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb4ebdc75b3a73a43da5ec.md (hunk 1 above)
Comment (usamabasharat) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb4ebdc75b3a73a43da5ec.md:423]: @jdwilkin4, Oh my bad, I misunderstood the issue. Let me fix.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb4ebdc75b3a73a43da5ec.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ assert.match(code, /\s*setTimeout\s*\(\s*\(\s*\)\s*=>[^,]*,\s*2000\s*\)/s);
 Your callback function should set the `checkpointScreen` `display` property to `none`.
 
 ```js
-assert.match(code, /\s*if\s*\(isCheckpointCollisionDetectionActive\)\s*{\s*setTimeout\s*\(\s*\(\s*\)\s*=>\s*\(\s*checkpointScreen\.style\.display\s*=\s*["']none["']\s*\)\s*,\s*2000\s*\)\s*;?\s*}/s);
+assert.match(code, /\s*if\s*\(isCheckpointCollisionDetectionActive\)\s*{\s*setTimeout\s*\(\s*function\s*\(\)\s*{\s*checkpointScreen\.style\.display\s*=\s*["']none["']\s*;\s*},\s*2000\s*\)\s*;?\s*}/s);
```
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb4ebdc75b3a73a43da5ec.md]:
```suggestion
assert.match(code, /\s*if\s*\(isCheckpointCollisionDetectionActive\)\s*{\s*setTimeout\s*\(\s*\(\s*\)\s*=>\s*(\(\s*checkpointScreen\.style\.display\s*=\s*("|')none\2\s*\)|\{\s*checkpointScreen\.style\.display\s*=\s*("|')none\3\s*\}|\s*checkpointScreen\.style\.display\s*=\s*("|')none\4\s*)\s*,\s*2000\s*\)\s*;?\s*}/s);
```
This suggestion matches:
- () => ( checkpointScreen.style.display = "none" )
- () => { checkpointScreen.style.display = "none" }
- () => checkpointScreen.style.display = "none"

Additionally ensures the quotes around `none` are matching.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb4ebdc75b3a73a43da5ec.md [hunk 3]
```diff
@@ -42,7 +42,7 @@ assert.match(code, /\s*setTimeout\s*\(\s*\(\s*\)\s*=>[^,]*,\s*2000\s*\)/s);
 Your callback function should set the `checkpointScreen` `display` property to `none`.
 
 ```js
-assert.match(code, /\s*if\s*\(isCheckpointCollisionDetectionActive\)\s*{\s*setTimeout\s*\(\s*\(\s*\)\s*=>\s*\(\s*checkpointScreen\.style\.display\s*=\s*["']none["']\s*\)\s*,\s*2000\s*\)\s*;?\s*}/s);
+assert.match(code, /\s*if\s*\(isCheckpointCollisionDetectionActive\)\s*{\s*setTimeout\s*\(\s*\(\s*\)\s*=>\s*(\(\s*checkpointScreen\.style\.display\s*=\s*("|')none\2\s*\)|\{\s*checkpointScreen\.style\.display\s*=\s*("|')none\3\s*\}|\s*checkpointScreen\.style\.display\s*=\s*("|')none\4\s*)\s*,\s*2000\s*\)\s*;?\s*}/s);
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb4ebdc75b3a73a43da5ec.md:45]: This still fails the option that was laid out in the issue 

```js
  if(isCheckpointCollisionDetectionActive) {
    setTimeout(() => { checkpointScreen.style.display = "none"; }, 2000);
  }
```
Screenshot 2024-01-04 at 8 30 27 AM

this should fix it though
Screenshot 2024-01-04 at 8 34 16 AM

```suggestion
assert.match(code, /\s*if\s*\(isCheckpointCollisionDetectionActive\)\s*{\s*setTimeout\s*\(\s*\(\s*\)\s*=>\s*(\(\s*checkpointScreen\.style\.display\s*=\s*("|')none\2\s*\)|\{\s*checkpointScreen\.style\.display\s*=\s*("|')none\3\s*;?\s*\}|\s*checkpointScreen\.style\.display\s*=\s*("|')none\4\s*)\s*,\s*2000\s*\)\s*;?\s*}/s);
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
With this change we should be good

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb4ebdc75b3a73a43da5ec.md (hunk 3 above)
Comment (gikf) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb4ebdc75b3a73a43da5ec.md:45]: I though I was copying the example directly when I was testing changed regex... Ah, the optional semicolon strucking when you expect it the least! :eyes:

**Review by gikf — APPROVED** [review]

**Review by jdwilkin4 — APPROVED** [review]

**Pull request merged**